### DISCUSSION TITLE
Repeated Puma Restart Issue with Kamal Deployment on DigitalOcean

### DISCUSSION THREAD

**Discussion by leegeng** [basecamp/kamal#853] [Q&A] [answered]
Hello everyone,

I am currently using Kamal(at that time, v1.6.0) to deploy a Rails application on DigitalOcean, and I have encountered an issue twice now. Unfortunately, I do not have precise steps to reproduce the problem, but here is what I've observed:

```ruby
=> Booting Puma
=> Rails 7.0.8.4 application starting in production
=> Run `bin/rails server --help` for more startup options
A server is already running. Check /app/tmp/pids/server.pid.
Exiting
```
This error message repeats, causing the CPU usage to spike significantly.
<img width="813" alt="image" src="https://github.com/basecamp/kamal/assets/343270/82c51989-8ab3-4b45-8984-f1bb9a7948a1">

This is an early-stage project, and the Puma configuration is set to its default values. I am wondering if anyone has experienced a similar issue.

From the logs, it seems that Puma is attempting to restart unnecessarily, failing to do so, and entering a loop that drives up CPU usage. I am not sure if this is due to a specific command being issued by Kamal during deployment. Before diving into the code, I wanted to check if there are any known issues or discussions related to this problem, as I couldn't find anything relevant in the existing issues or discussions.

Is there anything else I should check or any additional steps I should take to diagnose this issue?

Thank you in advance for your help!

Best regards,
Kyoungwon Lee

**Comment by acidtib** [comment, marked as answer]
ive seen a couple people fix this issue by removing the pid if it exists, on your docker-entrypoint add this

```
if [ -f tmp/pids/server.pid ]; then
  rm tmp/pids/server.pid
fi
```

**Reply by leegeng** [reply]
@acidtib 
Thank you for the response. However, it doesn't entirely make sense to me to explicitly remove the server.pid within the Docker entrypoint, it seems like it could be a viable solution to the problem.

**Reply by nickhammond** [reply]
Here's a more extensive thread on this https://github.com/docker/compose/issues/1393
